### ISSUE TITLE
SingleApplication::instanceStarted signal won't trigger in random order

### ISSUE THREAD

**Issue by mehranmd** [itay-grudev/SingleApplication#86]
Hi and Thanks for the source code sharing.
I used this source as a submodule in my Qt/Quick application to block the second instance of the application and raise the previously hided ApplicationWindow(qml class).
so naturally replaced the QApplication with QGuiApplication in qmake:
`include(singleapplication/singleapplication.pri) `
`DEFINES += QAPPLICATION_CLASS=QGuiApplication`
and created a signal in one of my QML registered classes.
`SingleApplication app( argc, argv, false, SingleApplication::Mode::User,100);`
`...`
`ControlCenter controlCenterObject;`
`QObject::connect(
        &app,
        &SingleApplication::instanceStarted,
        &controlCenterObject,
        &ControlCenter::raisWindow
    );`
and if your familiar with the QML, created a Connection to receive the signal:
`Connections{
        target: ControlCenter
        onRaisWindow:{
            console.log("Requested to raise the window")
            window.show()
            window.raise()
            window.requestActivate()
        }
    }`
almost everything works fine but small problem is that it's not working every time i start a secondry instance of application.
I tracked down the issue and found out that the signal won't trigger from the SingleApplication.
Any idea what's wrong?

**Comment by mehranmd** [comment]
More information:
Host os: Windows 10 (latest updates), Windows 7 (VMWare and latest updates)
Qt version: 5.12.3 dynamic build
Compiler: MinGW 7.3 32-bit

**Comment by itay-grudev** [comment]
I just saw this issue. I apologize for not replying earlier. Are you still having the same issue? Do you have secondary instance support enabled or just primary + instanceStarted?

**Comment by fogelton** [comment]
I have the same problem, randomly maybe 20% of the time the signal instanceStarted is not triggered and the primary app is not opened (slot connected to the signal). The secondary instance support is disabled in my case. I just wanted one app running at a time. Using qt 5.15 msvc2019, win 10 home latest,
I use these paramateres
explicit SingleApplication( int &argc, char *argv[], bool allowSecondary = false, Options options = Mode::SecondaryNotification, int timeout = 1000 );

**Comment by itay-grudev** [comment]
@fogelton You shouldn't need `Options options = Mode::SecondaryNotification` unless you've set `allowSecondary = true`. Although setting it shouldn't interfere in non-secondary instance cases. I'll check the code if thats the case. In the mean time can you try it without the flag?

**Comment by fogelton** [comment]
I changed it to user, the behavior is the same, about 20% not triggered the signal.

**Comment by itay-grudev** [comment]
I identified the issue. It occurs when you run instances in a very rapid succession, like:

```bash
./basic &; ./basic &; ./basic &; ./basic &; sleep 1; killall basic
```

The reason is that there wasn't enough time for the `QLocalServer` to get initialised. Every attempt after enough time has passed (milliseconds) works fine.

My initial implementation of the internal messaging doesn't account for this scenario. I'm working on a fix.

**Comment by itay-grudev** [comment]
Fixed in [`v3.1.5`](https://github.com/itay-grudev/SingleApplication/releases/tag/v3.1.5).
